store/tikv/snapshot.go

```diff
@@ -174,7 +175,23 @@ func (s *tikvSnapshot) BatchGet(ctx context.Context, keys []kv.Key) (map[string]
 		s.mu.cached = make(map[string][]byte, len(m))
 	}
 	for _, key := range keys {
-		s.mu.cached[string(key)] = m[string(key)]
+		val := m[string(key)]
+		s.mu.cachedSize += len(key) + len(val)
+		s.mu.cached[string(key)] = val
+	}
+
+	const cachedSizeLimit = 10 << 30
```

Comment:
We could change it to a const for https://github.com/pingcap/tidb/blob/9af236c1b7f8f8050e9c163a78b83ec4564b9e2e/config/config.go#L965 and this one in another PR.